### PR TITLE
Update topics for front-end questions

### PR DESCRIPTION
**Pull request by meiti-x**

### CODE REVIEW

File: frontend/general/categories/css.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
 # CSS
 
-* Box-Model
\ No newline at end of file
+* Box-Model
+* BEM
+* Flex-Box
```
Comment (Hossein13M) [inline frontend/general/categories/css.md:5]: If we do not intend to include Sass questions in a different section, we can also add some questions about Sass and postCSS.

File: frontend/general/categories/typescript.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
 # Typescript
 
-* Type
+* Basic Types
+* Type Inference
 * Interface
+* Union
+* Enum
+* Generics
```
Comment (Hossein13M) [inline frontend/general/categories/typescript.md:8]: Decorators, Type Assertion could fit here.

**Review by Hossein13M — APPROVED** [review]
We can extend this categories in the future, but for now, the list seems OK.

**Review by Hossein13M — APPROVED** [review]
Thanks for your contribution @mahdi-momeni 
It seems OK.

**Review by MoienTajik — APPROVED** [review]

**Pull request merged**